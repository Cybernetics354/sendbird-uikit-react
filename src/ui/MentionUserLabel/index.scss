@import '../../styles/variables';

.sendbird-mention-user-label {
  &:hover {
    cursor: pointer;
  }
  display: inline-block;
  font-family: var(--sendbird-font-family-default);
  font-size: 14px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.43;
  letter-spacing: normal;
  @include themed() {
    color: t(on-bg-1);
  }
  &.reverse {
    @include themed() {
      color: t(on-content-1);
    }
  }
  &.purple {
    font-family: Roboto;
    font-size: 18px;
    font-weight: bold;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: normal;
    @include themed() {
      color: t(primary-3);
    }
  }
}
