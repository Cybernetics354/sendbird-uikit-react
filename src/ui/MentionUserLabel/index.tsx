import React from 'react';
import './index.scss';

interface MentionUserLabelProps {
  className?: string
  children?: string;
  isReverse?: boolean;
  color?: string;
}

export default function MentionUserLabel({
  className = '',
  children,
  isReverse = false,
  color,
}: MentionUserLabelProps): JSX.Element {
  return (
    <span
      className={`sendbird-mention-user-label ${className} ${isReverse ? 'reverse' : ''} ${color}`}
    >
      {children}
    </span>
  );
}
