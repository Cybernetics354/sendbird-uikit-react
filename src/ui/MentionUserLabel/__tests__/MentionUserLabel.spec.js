import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import MentionUserLabel from "../index";

describe('MentionUserLabel', () => {
  it('should do a snapshot test of the MentionUserLabel DOM', function() {
    const text = "example-text";
    const component = renderer.create(
      <MentionUserLabel />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
