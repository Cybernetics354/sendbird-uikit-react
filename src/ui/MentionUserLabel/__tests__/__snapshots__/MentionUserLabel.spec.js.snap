// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MentionUserLabel should do a snapshot test of the MentionUserLabel DOM 1`] = `<div />`;
